File: Easy/1518. Water Bottles/solution.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = 0
        additional = numBottles
        while numBottles >= numExchange:
            res += additional
            additional = numBottles // numExchange
            emptyBottles = numBottles % numExchange
            numBottles = additional + emptyBottles
        return res + additional

File: Easy/1518. Water Bottles/solution.cs
public class Solution
{
    public int NumWaterBottles(int numBottles, int numExchange)
    {
        int res = 0;
        int additional = numBottles;
        while (numBottles >= numExchange)
        {
            res += additional;
            additional = numBottles / numExchange;
            int emptyBottles = numBottles % numExchange;
            numBottles = additional + emptyBottles;
        }
        return res + additional;
    }
}